src/compiler/checker.ts

```diff
@@ -6240,6 +6240,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                 }
                 let annotationType = getTypeFromTypeNodeWithoutContext(existing);
+                if (isErrorType(annotationType)) {
+                    // allow "reusing" type nodes that resolve to error types
+                    // those can't truly be reused but it prevents cascading errors in isolatedDeclarations
+                    // for source with errors there is no guarantee to emit correct code anyway
+                    return true;
+                }

```

Comment:
This could be handled elsewhere but this place is an easy and effective fix.

Note that in a sense this was already happening - just in different situations. Consider this:
```ts
export const foo3 = (type: Unresolved): void => {};
```

In here both the `annotationType` and `type` are error types, they just happen to be the same aliased error types. So they are assumed to be the same on the basis of `typeFromTypeNode === type` in `typeNodeIsEquivalentToType`.

The reported case is the same yet different. Both types are also error types there but the `type` is a regular non-aliased error type whereas the `annotationType` is an aliased error type. Based on that `typeNodeIsEquivalentToType` doesn't assume that they are the same and thus `typeFromParameter` goes into `inferTypeOfDeclaration` which leads to reporting the error.

Reply:
Is there a condition in which this will cause us to try and print something unprintable?

Or will it not matter because it'll be an error anyway?

Reply:
I propose this fix mainly on the assumption of "will it not matter because it'll be an error anyway". It seems to me that if the error is there anyway then any emit is basically a fair game. It should reuse the annotation node - so at the very least it should just be broken in the same way.